packages/core/src/StateNode.ts

```diff
@@ -614,8 +614,8 @@ class StateNode<
    * @param state The state value or State instance
    */
   public getStateNodes(
-    state: StateValue | State<TContext, TEvent>
-  ): Array<StateNode<TContext, any, TEvent>> {
+    state: StateValue | State<TContext, TEvent, any, TTypestate>
+  ): Array<StateNode<TContext, any, TEvent, TTypestate>> {
```

Comment:
actually maybe this shouldn't be parametrized with `TTypestate` as the "correct" value is only relevant for State and such as it's a top-level value and this one returns nested StateNodes as well, thoughts?

Reply:
Yeah, we don't need it here.

Reply:
k, gonna adjust later today